es6-classes/stopwatch.js

```diff
@@ -0,0 +1,29 @@
+/* exported Stopwatch */
+
+class Stopwatch {
+  constructor(startTimeInSeconds) {
+    this.elapsedSeconds = startTimeInSeconds;
+  }
+}
+
+Stopwatch.prototype.tick = function () {
+  this.elapsedSeconds++;
+};
+
+Stopwatch.prototype.getTime = function () {
+  const secondsPerHour = 3600;
+  const secondsPerMinute = 60;
+  let seconds = this.elapsedSeconds;
+  let hours = Math.floor(seconds / secondsPerHour);
+  seconds -= (secondsPerHour * hours);
+  let minutes = Math.floor(seconds / secondsPerMinute);
+  seconds -= (secondsPerMinute * minutes);
+  hours = hours.toString().padStart(2, '0');
+  minutes = minutes.toString().padStart(2, '0');
+  seconds = seconds.toString().padStart(2, '0');
+  return `${hours}:${minutes}:${seconds}`;
+};
+
+Stopwatch.prototype.reset = function () {
+  this.elapsedSeconds = 0;
+};
```

Comment:
With classes we don't prototype the methods on in this way, we define them directly within the class' code block. When they are defined in the classes code blog they will be automatically prototyped onto the class. For example:

```JAVASCRIPT
class DoSomething {
  constructor(thing) {
    this.thing = thing;
  }
  
  aMethod() {
  
  }

  bMethod() {
  
  }
  
  cMethod() {
  
  }
}
```